### ISSUE TITLE
uninstalling and reinstalling doesn't fully reset Gossip ( windows 10 )

### ISSUE THREAD

**Issue by GretaFrank** [mikedilger/gossip#632]
have been away from Gossip for a while using noStrudel but recently with the whole switch to read and write relays noStrudel is pretty banged up and not really usable so i'm trying to get Gossip to work

i uninstalled the old one and installed the latest one to Windows 10 using MSI file but relays all time out

what bothers me is when i uninstall and reinstall it only asks me for password ( Gossip calls it "passphrase" even though it's clearly just a password ) - it is storing my account info somewhere even when uninstalled.

i did try deleting the the account manually from the Keys menu ( Gossip calls it "delete this identity" ) but even though that does require me to input my NSEC all over again it somehow still shows the bio i used to have months ago back when Gossip used to work for me and not the BIO i currently use which all other clients show.

so a bunch of information related to my account and / or settings is being stored somewhere by Gossip and i don't know how to reset it.

i want to reset everything COMPLETELY and start from scratch.

my real goal is of course just to get relays to connect and Gossip to work but i feel like the prudent thing to do in terms of troubleshooting is a clean, fresh start with nothing but my NSEC and with all settings and any other information wiped.

HELP !

**Comment by mikedilger** [comment]
Your gossip configuration is stored in your gossip directory which is a subdirectory of your data_dir defined here:
`https://docs.rs/dirs/latest/dirs/fn.data_dir.html`

So something like (use your drive letter and username):  `C:\Users\Alice\AppData\Roaming\gossip\`

You can nuke that entire directory.

When you restart you'll find a new wizard to help get you setup. Let me know if it doesn't work.

**Comment by mikedilger** [comment]
The term "passphrase" is used in order to encourage users to use longer passwords.

Once you are running, please go to the menu "Relays > Active Relays" at let me know if there are any with green circles, or if they are all showing yellow circles (or if there is nothing in the list).  I've heard from one other person that none of the relays were connecting, so I'd be keen to figure out why since I doesn't happen to most people.

**Comment by GretaFrank** [comment]
first of all the other person who complained about relays not connecting is me ...

now ... i uninstalled again and deleted the directory you mentioned, then reinstalled the latest version from MSI

i logged in with my NSEC and created a new password.  in the keys page it correctly recovered my NPUB.

everything is blank now - my entire profile is blank - no name or bio - the relay list is blank ( except purplepages ) and the list of people i'm following is empty ...

i added relay.mostr.pub and restarted gossip 

it still says "timeout, retry in ( countdown timer ) seconds" for both purplepages and mostr.pub just as it previously would say for all the relays ... the circles next to relays are YELLOW

so yeah the other person is me so maybe it only happens to me, but that's what i'm getting ...

![image](https://github.com/mikedilger/gossip/assets/146442815/9c96bb21-c497-4f37-a10b-0382f8a56ede)

![image](https://github.com/mikedilger/gossip/assets/146442815/20843708-c752-4229-bff6-2def33187f86)

**Comment by GretaFrank** [comment]
so in screen shot above one circle is green, but it just flashes green for a second saying "connected" once a minute then goes back to yellow retry ...

**Comment by GretaFrank** [comment]
trying to post a test note gave no warning and it shows up in my notes, but doesn't actually post when viewed from other clients

![image](https://github.com/mikedilger/gossip/assets/146442815/9bf4b0a0-bf14-4648-92f5-8809f89d8c20)

**Comment by mikedilger** [comment]
Are you running it from a console window?  Do you have the console output?  There might be a message in there as to what precisely is happening.

The cause is not obvious.  Is there a firewall in the way?  I dunno.

When posting a test note it will remember that note locally even if it never sees it from a relay. If you hover over the eye symbol on the right side of the note it shows a list of relays the note was seen on.  For you that list is probably empty.

**Comment by GretaFrank** [comment]
console output after logging in with my "passphrase"

←[2m2024-01-30T00:55:27.536901Z←[0m ←[32m INFO←[0m ←[2mC:\Users\Mike\.cargo\git\checkouts\gossip-relay-picker-6fa537638e7e06ab\360aadc\src\lib.rs←[0m←[2m:←[0m←[2m268:←[0m wss://relay.mostr.pub/ goes into the penalty box for 60 seconds until 1706576187
←[2m2024-01-30T00:55:27.542539Z←[0m ←[32m INFO←[0m ←[2mgossip-lib\src\overlord\mod.rs←[0m←[2m:←[0m←[2m480:←[0m Minion wss://relay.mostr.pub/ will restart in 60 seconds to continue persistent jobs
←[2m2024-01-30T00:55:27.778158Z←[0m ←[31mERROR←[0m ←[2mgossip-lib\src\overlord\mod.rs←[0m←[2m:←[0m←[2m408:←[0m Minion wss://purplepag.es/ completed with error: Websocket: TLS error: rustls error: invalid peer certificate: BadEncoding
←[2m2024-01-30T00:55:27.779467Z←[0m ←[32m INFO←[0m ←[2mC:\Users\Mike\.cargo\git\checkouts\gossip-relay-picker-6fa537638e7e06ab\360aadc\src\lib.rs←[0m←[2m:←[0m←[2m268:←[0m wss://purplepag.es/ goes into the penalty box for 60 seconds until 1706576187
←[2m2024-01-30T00:55:27.783774Z←[0m ←[32m INFO←[0m ←[2mgossip-lib\src\overlord\mod.rs←[0m←[2m:←[0m←[2m480:←[0m Minion wss://purplepag.es/ will restart in 60 seconds to continue persistent jobs

**Comment by mikedilger** [comment]
There is some TLS problem.  If you compile, you can use a different TLS setup.  Compiling isn't very difficult but you have to install git and rust:

1) Install git for windows https://gitforwindows.org/
2) Install rust according to simple instructions here: https://rustup.rs/
3) right click and start a git shell, then in that shell
4) git clone https://github.com/mikedilger/gossip
5) cd gossip
6) cargo build --release --features="native-tls"
7) Run it from same shell for now to see if it works:  "./target/release/gossip"

**Comment by GretaFrank** [comment]
i'm too stupid for that sort of solution, but thanks anyway.

maybe i will attempt it some day if i'm bored.

i will go now LOL

**Comment by GretaFrank** [comment]
you know at first i posted just the lines where it was talking about relays and associated errors ... but then i decided to just post the whole thing ...

and i notice it keeps talking about Razer ( i have Razer mouse ) ... i think Razer software killed my entire Windows on my other PC ... it killed it like THREE TIMES but the first two times i was able to do repair, the third time i had to reinstall windows ...

when i see it repeating Razer Chroma SDK over and over and over i get nervous LOL

**Comment by mikedilger** [comment]
Maybe this Razor Chroma thing added a root certificate to your local certificate store that is somehow corrupt, and now all your SSL connections don't work.

**Comment by bu5hm4nn** [comment]
The title of the issue suggests a feature request to offer the user a mechanism of wiping the data directory. Should we make that a feature request? @dtonon @mikedilger

**Comment by mikedilger** [comment]
I'll open a separate issue for that.

**Comment by mikedilger** [comment]
Yes we figured out that the combination of using rustls with system certificates was highly unreliable because rustls is very strict.

**Comment by GretaFrank** [comment]
good news Frens !  it works now !

when the new update came out a few hours ago i used windows uninstall tool for the old version, then installed the new one and even though it didn't seem to throw the previous error codes and now seemed to connect to some of the relays it still wouldn't show the timeline.

i then uninstalled again but this time also deleted the roaming\appdata folder and went through the install process from scratch importing NSEC and setting up a password - and this time it worked.

i see the timeline, the notifications and i can post.

time will tell how well it actually works but at least it works finally.

thanks everyone !
